src/WinUIEx/WebAuthenticator.cs

```diff
@@ -107,7 +109,17 @@ private static bool IsUriProtocolDeclared(string scheme)
             {
                 try
                 {
-                    var jsonObject = System.Text.Json.Nodes.JsonObject.Parse(state) as JsonObject;
+                    JsonObject? jsonObject;
+
+                    try
+                    {
+                        jsonObject = System.Text.Json.Nodes.JsonObject.Parse(state) as JsonObject;
+                    } 
+                    catch (Exception e)
+                    {
+                        jsonObject = System.Text.Json.Nodes.JsonObject.Parse(Uri.UnescapeDataString(state)) as JsonObject;
+                    }
+
```

Comment:
Currently a pretty weird way to support state parameter with and without escaping. I know escaping got introduced before, and reverted again. So curious if this should be something behind a feature flag?

However, even thought this is a kinda weird way, it should never hit the catch for those already using it today.